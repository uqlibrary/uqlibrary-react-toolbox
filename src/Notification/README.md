# Notification

simple reusable notification panel for UQ Library applications using ReactJs

## Props
- `notifcationDetails: PropTypes.object`
- `duration: PropTypes.number`
    
## Usage``

**./component/App.js**
```jsx
import React from 'react';
...

import {Notification} from 'uqlibrary-react-toolbox';
  
  :

showNotification = () => {
   this.props.loadNotification('This is a test message');
}

const App = () => (
  <div>
    <Notification />
    <RaisedButton
              onTouchTap={this.showNotification}
              label="Test Button"
            />
  </div>
);

```

**./container/App.js**
```
import {connect} from 'react-redux';
import {loadNotification} from 'uqlibrary-react-toolbox';

AppContainer = connect(null, dispatch => {
    return {
        loadNotification: (message) => dispatch(loadNotification(message))
    };
})(App);

export default AppContainer;
```



**./SomeComponent.js**
```
ReactDOM.render(
  <App />,
  document.getElementById('app')
);
```