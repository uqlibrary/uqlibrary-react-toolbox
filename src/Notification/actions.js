// Types
export const NOTIFICATION_DISPLAYED = 'NOTIFICATION_DISPLAYED';

/**
 * Displayed the notification with the custom message
 * @returns {{type: string, payload: string}}
 */
export function loadNotification(message) {
    return {
        type: NOTIFICATION_DISPLAYED,
        payload: {
            isOpen: true,
            message: message
        }
    };
}
