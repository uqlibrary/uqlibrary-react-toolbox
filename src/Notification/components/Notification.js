import React from 'react';
import Snackbar from 'material-ui/Snackbar';
import PropTypes from 'prop-types';

export default class Notification extends React.Component {

    static propTypes = {
        notifcationDetails: PropTypes.object,
        duration: PropTypes.number
    };

    static defaultProps = {
        duration: 4000
    }

    constructor(props) {
        super(props);
    }

    render() {
        return (
            <Snackbar
                className="notification"
                open={this.props.notifcationDetails.get('isOpen') || false}
                message={this.props.notifcationDetails.get('message') || ''}
                autoHideDuration={this.props.duration}
            />
        );
    }
}
