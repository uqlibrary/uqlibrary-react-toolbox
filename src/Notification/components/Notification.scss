@import '../../../sass/variables';

// Styles for the notification panel
.notification {
  background-color: $theme-color;
}
