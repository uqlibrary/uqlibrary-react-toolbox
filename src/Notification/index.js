export {default as Notification} from './containers/Notification';
export {default as notificationReducer} from './reducer';
