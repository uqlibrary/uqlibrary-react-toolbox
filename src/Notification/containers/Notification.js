import {connect} from 'react-redux';
import Notification from '../components/Notification';

const NotificationContainer = connect((state) => {
    return {
        notifcationDetails: state.get('notification').get('notificationList')
    };
})(Notification);

export default NotificationContainer;
