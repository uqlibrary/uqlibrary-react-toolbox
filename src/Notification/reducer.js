import Immutable from 'immutable';

import {NOTIFICATION_DISPLAYED} from './actions';

// Immutable state
export const initialState = Immutable.fromJS({
    notificationList: {}
});

const notificationReducer = (state = initialState, action) => {
    switch (action.type) {
        case NOTIFICATION_DISPLAYED:
            return state.set('notificationList', Immutable.fromJS(action.payload));
        default:
            return state;
    }
};

export default notificationReducer;
