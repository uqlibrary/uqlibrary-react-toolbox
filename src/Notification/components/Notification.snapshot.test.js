jest.dontMock('./Notification');

import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Immutable from 'immutable';

import Notification from './Notification';

function setup() {
    const props = {
        notifcationDetails: Immutable.fromJS({
            isOpen: true,
            message: 'test message'
        })
    }

    return shallow(<Notification {...props} />);
}


describe('Notification snapshot tests', () => {
    it('renders the notificiation component', () => {
        const app = setup();
        expect(toJson(app)).toMatchSnapshot();
    });
});
