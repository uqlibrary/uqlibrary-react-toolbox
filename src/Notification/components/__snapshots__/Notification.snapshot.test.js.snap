// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification snapshot tests renders the notificiation component 1`] = `
<Snackbar
  autoHideDuration={4000}
  className="notification"
  message="test message"
  open={true}
/>
`;
